From a7dfb9ad262f2f11b0d5b864ecb30b47392f5b01 Mon Sep 17 00:00:00 2001
From: Feng Tang <feng.tang@intel.com>
Date: Mon, 4 Jun 2018 15:17:42 +0800
Subject: [PATCH 1581/2345] mfd: intel-lpss: Make driver probe asynchronous

LPSS driver's probe takes quite some time, around 50ms to 100ms
on Apollo Lake platform for all the spi/uart/i2c controllers. Making
the driver probe async can help to reduce the boot time.

Signed-off-by: Feng Tang <feng.tang@intel.com>
Cc: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Cc: Mika Westerberg <mika.westerberg@linux.intel.com>
Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
---
 drivers/mfd/intel-lpss-acpi.c | 1 +
 drivers/mfd/intel-lpss-pci.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/mfd/intel-lpss-acpi.c b/drivers/mfd/intel-lpss-acpi.c
index 7911b0a..dd655d4 100644
--- a/drivers/mfd/intel-lpss-acpi.c
+++ b/drivers/mfd/intel-lpss-acpi.c
@@ -116,6 +116,7 @@ static struct platform_driver intel_lpss_acpi_driver = {
 		.name = "intel-lpss",
 		.acpi_match_table = intel_lpss_acpi_ids,
 		.pm = &intel_lpss_acpi_pm_ops,
+		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
 	},
 };
 
diff --git a/drivers/mfd/intel-lpss-pci.c b/drivers/mfd/intel-lpss-pci.c
index d9ae983..ce4ca4b 100644
--- a/drivers/mfd/intel-lpss-pci.c
+++ b/drivers/mfd/intel-lpss-pci.c
@@ -260,6 +260,7 @@ static struct pci_driver intel_lpss_pci_driver = {
 	.remove = intel_lpss_pci_remove,
 	.driver = {
 		.pm = &intel_lpss_pci_pm_ops,
+		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
 	},
 };
 
-- 
2.7.4

